packages/sdk/src/adapters/multipeer/session.ts

```diff
@@ -68,79 +69,87 @@ export class Session extends EventEmitter {
     /**
      * Performs handshake and sync with the app
      */
-    public connect(): Promise<void> {
-        return new Promise<void>((resolve, reject) => {
-            // Start with Handshake protocol
+    public async connect() {
+        try {
             const handshake = this._protocol = new SessionHandshake(this);
-            handshake.on('protocol.handshake-complete', () => {
-                // Sync app state to session
-                const sync = this._protocol = new SessionSync(this);
-                sync.on('protocol.sync-complete', () => {
-                    // Switch to Execution protocol
-                    const execution = this._protocol = new SessionExecution(this);
-                    execution.on('recv', message => this.recvFromApp(message));
-                    execution.startListening();
-                    resolve();
-                });
-                sync.startListening();
-            });
-            handshake.startListening();
-        });
+            await handshake.run();
+
+            const sync = this._protocol = new SessionSync(this);
+            await sync.run();
+
+            const execution = this._protocol = new SessionExecution(this);
+            execution.on('recv', message => this.recvFromApp(message));
+            execution.startListening();
+        } catch (e) {
+            log.error('network', e);
+            this.disconnect();
+        }
     }
 
     public disconnect() {
-        this._conn.off('close', this.disconnect);
-        this._conn.off('error', this.disconnect);
-        this._conn.close();
-        this.emit('close');
+        try {
+            this._conn.off('close', this.disconnect);
+            this._conn.off('error', this.disconnect);
+            this._conn.close();
+            this.emit('close');
+        } catch { }

```

Comment:
>catch { } [](start = 10, length = 9)

should you log error here?

Reply:
I'm intentionally not logging errors here because we're in the process of tearing down the session. Various errors can happen depending on how "started up" the session was in the first place. I just want to make sure exceptions don't interrupt the shutdown process from completing.

---
In reply to: [252832190](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/155#discussion_r252832190) [](ancestors = 252832190)